src/com/webonise/carbuilder/CompanyFactory.java

```diff
@@ -0,0 +1,40 @@
+package com.webonise.carbuilder;
+
+public class CompanyFactory extends AbstractCarFactory{
+
+	@Override
+	public ITypeOfCar getTypeOfCar(String TypeOfCar) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public ICompany getCompany(String Company) {
+		if(Company == null){
+	         return null;
+	      }		
+	      
+	      if(Company.equalsIgnoreCase("BMW")){
+	         return new BMW();
+	         
+	      }else if(Company.equalsIgnoreCase("Audi")){
+	         return new Audi();
+	         
+	      }else if(Company.equalsIgnoreCase("Mercedes")){
+	         return new Mercedes();
+	      }
```

Comment:
We can replace this if-else with switch.
That's why Enums are preferable. If you missed out one of the Enum case it will give you a warning.